### PR TITLE
openQA test fails

### PR DESCRIPTION
**Pull request by varkoly**
## Problem

The new feature XS_ROUTIN_MODE coused some problems in autoinstallation.
Reason was some typo and bad logic.

patterns-sap-hana is not available for SLES. bone-installation-wizard is desined for SLES.

- *https://bugzilla.suse.com/show_bug.cgi?id=1220851*
- *https://openqa.suse.de/tests/13548816*

## Solution

*Fix auto.rb module an hana_inst.sh script.*
*Remove bad requirement.*

### CODE REVIEW

File: package/sap-installation-wizard.spec [hunk 1]
```diff
@@ -60,13 +60,15 @@ Authors:
...
 Group:          System/YaST
-Version:        5.0.1
+Version:        5.0.2
 Release:        0
 PreReq:         /bin/mkdir %fillup_prereq yast2
 BuildRequires:  yast2
 BuildRequires:  sapconf
 Requires:       autoyast2
 Requires:       autoyast2-installation
+Requires:       patterns-sap-hana
```
Comment (jreidinger) [inline package/sap-installation-wizard.spec:70]: to be honest I do not get logic. For SP6 this dependency was removed and here it was added.

File: src/bin/hana_inst.sh [hunk 2]
```diff
@@ -254,23 +255,12 @@ hana_lcm_workflow()
...
-       echo -e "db_mode=multidb\ndb_isolation=high\n"  > ${MEDIA_TARGET}/hana_mdc.conf
-     ;;
-   esac
    cd "${HDBLCMDIR}"
    TOIGNORE="check_signature_file"
    if [ -e /root/hana-install-ignore ]; then
      TOIGNORE=$(cat /root/hana-install-ignore)
    fi
-   if [ -e ${MEDIA_TARGET}/hana_mdc.conf ]; then
+   if [ ${XS_ROUTING_MODE} == "ports" ]; then
```
Comment (jreidinger) [inline src/bin/hana_inst.sh:263]: same here. For SP6 the condition was changed to handle also empty variable cases and here it is not case.

File: src/clients/inst_sap-start.rb [hunk 3]
```diff
@@ -65,7 +74,7 @@ def main
       ret
     end
 
-    def constumize_sap_installation(start_wizard, start_rdp)
+    def costumize_sap_installation(start_wizard, start_rdp)
```
Comment (jreidinger) [inline src/clients/inst_sap-start.rb:77]: and here typo "costumize" instead of "customize"

File: src/lib/y2sap/auto.rb [hunk 4]
```diff
@@ -175,12 +175,25 @@ def prepare_hana(prod)
...
+        File.write(@inst_dir + "/ay_q_virt_hostname", prod["sapVirtHostname"])
+      end
+      xs_routing_mode = prod.key?("xsRoutingMode") ? prod.key?("xsRoutingMode") : "ports"
+      if !["hostname", "ports"].include?(xs_routing_mode)
+        Yast::Popup.Error("Bad XS routing mode: #{xs_routing_mode}. Provided values are 'hostname' or 'ports'")
+        @error = true
+        return :abort
+      end
+      if xs_routing_mode == "hostname" && !prod.key?("xsDomainName")
+        Yast::Popup.Error("If XS routing mode is set to hostname you have to define xsDomainName also")
```
Comment (jreidinger) [inline src/lib/y2sap/auto.rb:191]: also here it looks like old SP6 code without new changes from https://github.com/yast/sap-installation-wizard/pull/38

Comment (varkoly) [reply src/lib/y2sap/auto.rb:191]: Sorry. I've found the problem.... 
Unfortunately I've merged SLE-15-SP5 instead of SLE-15-SP6 :-(

Comment (varkoly) [reply src/lib/y2sap/auto.rb]: Now I'v merged SP6 into master.
Thank you for your attention!

**Review by jreidinger — APPROVED** [review]

**Pull request merged**